### DISCUSSION TITLE
plot_all_topologies() function name

### DISCUSSION THREAD

**Discussion by michaelchin** [GPlates/gplately#207] [Ideas]
It seems to me that the function plot_all_topologies() plots topological boundaries + networks. The function name seems to me a bit ambiguous. Should we add new functions like plot_topological_boundaries() and plot_topological_networks(), and deprecate plot_all_topologies()?

**Comment by jcannon-gplates** [comment]
I guess there's also topological *lines*, but they don't tend to get plotted (since typically used as sections to build boundaries and networks from). So `plot_all_topologies()` is probably fine, and keeping it saves users from having to update their code.

Adding `plot_topological_boundaries()` and `plot_topological_networks()` sounds fine. Or could add `plot_topologies()` that takes flags for which topologies to plot. Probably what you suggested is better (more explicit).

**Comment by michaelchin** [comment]
Yes, let's keep `plot_all_topologies()`.

And I think we need `plot_topological_boundaries()` because if we use `plot_all_topologies()`, the map looks like the screenshot below.

<img width="832" alt="Screenshot 2024-06-27 at 1 47 40 PM" src="https://github.com/GPlates/gplately/assets/2688316/9f10ec7e-565e-4359-9a9a-f33045d755cd">

In GPlates, the plate boundaries look like this.

<img width="748" alt="Screenshot 2024-06-27 at 1 51 45 PM" src="https://github.com/GPlates/gplately/assets/2688316/cb665687-72c0-4d43-b846-35a3426f71c8">

And boundaries + networks look like this.

<img width="751" alt="Screenshot 2024-06-27 at 1 52 10 PM" src="https://github.com/GPlates/gplately/assets/2688316/f3119a1c-1ad3-462e-8660-d157f4850d57">

Maybe it is good to separate boundaries and networks. And should we plot the network mesh as well in gplately?

**Comment by michaelchin** [comment]
And I am not sure if we could plot networks properly in gplately. See this complex structure below.

<img width="701" alt="Screenshot 2024-06-27 at 2 03 26 PM" src="https://github.com/GPlates/gplately/assets/2688316/49a86e30-aefe-4b87-b9b0-1d137464b280">

**Reply by jcannon-gplates** [reply]
True, that screenshot has interior rigid blocks within the deforming region.

And I'm not sure you'd want to reproduce all the options available in GPlates, such as colouring by dilatation strain rate (filled)...

![image](https://github.com/GPlates/gplately/assets/16912600/3adc667a-ecf2-493e-b7c3-76a463f82dd4)

**Reply by jcannon-gplates** [reply]
I suppose you could still have something that distinguishes between a rigid plate and a deforming network. In which case plotting the mesh wireframe as a single colour could work - though pyGPlates doesn't expose that just yet (but will). I don't think the interior rigid blocks are exposed yet either, but will be.

**Comment by michaelchin** [comment]
In fact, I am not entirely sure if the map produced by `plot_all_topologies()` makes sense without network mesh.

**Reply by jcannon-gplates** [reply]
I think the newer topological models (like what's in GPlates 2.5) have networks that **overlay** on top of the rigid plates - so not drawing the networks just exposes the underlying rigid plates (ie, no holes). Some older models don't - so removing the networks leaves holes underneath. There was a bit of discussion about that [here](https://discourse.gplates.org/t/returning-geometry-of-larger-plates-in-pygplates-or-gplately/851/4?u=john.cannon).

There's also the issue of drawing **inactive** meshes (mentioned in that link) - whether the user wants to see meshes for deforming regions that are no longer active (at the current reconstruction time).

**Comment by michaelchin** [comment]
Thanks John. Now I see that this matter is much more complex than I thought. Let's keep the `plot_all_topologies()` function and plot the boundaries and networks together.

**Comment by michaelchin** [comment]
Just wonder if anyone would like to plot boundaries and networks in different colours, just to tell which are boundaries and which are networks. Since no users had brought this up, I guess it is obvious for researchers which are which or it is not important to distinguish boundaries from networks.
